code/ecdsap256/Hacl.P256.fsti

```diff
@@ -348,10 +353,15 @@ val compression_compressed_form: b: lbuffer uint8 (size 64) -> result: compresse
   )
 
 
-[@ (Comment " Input: result: uint8[64], \n scalar: uint8[32].
-  \n Output: bool, where True stands for the correct key generation. 
-  \n False means that an error has occurred (possibly that the result respresents point at infinity). 
-  ")]
+[@ (Comment "Convert a private key into an uncompressed public key.
+
+  Input: `scalar`, the private key, of type `uint8[32]`.
+  Output: `result`, the public key, of type `uint8[64]`.
```

Comment:
So this sounds like this isn't prefixed with the format (0x04)?
This should be made consistent with the compressed form. But for the comment we should mention what's in these 64 bytes.

Reply:
Then say this is the concatenation of `r`and `s`?

Reply:
This file introduces a terminology of "compressed (33), not compressed (65) and internal also called uncompress (64)". See comment earlier in the file. 

The OCaml API has a different take, see here: https://hacl-star.github.io/ocaml_doc/hacl-star/Hacl_star/Hacl/P256/index.html#points

I could try to align this comment to what is done in the OCaml API, and then maybe fix up the earlier introduction of terminology? Would that help? I'm trying to steer away from a complete overhaul of the documentation of this file (even though I'm quite tempted...).

Reply:
Officially the 65 bytes (including the 0x04) is the uncompressed form (compared to the compressed form) - see RFC 5480 or SEC1. So I think using "not compressed" is a little confusing. The 64-byte version (concatenation of `r` and `s`) is usually referred to as raw. But there's no official term as it's not part of any standard afaik.
(The API really shouldn't work on the raw values, but I think that's a more general issue and not for this comment.)

Reply:
I updated the documentation to refer to `raw`, `compressed` and `uncompressed`, of size 64, 33 and 65 respectively. This should hopefully clarify.